### ISSUE TITLE
[BUG] Fatal error in PMPI_Wait error running PACKAGES/drude/swm4-ndp/in.swm4-ndp.nh example with mpirun

### ISSUE THREAD

**Issue by mdelyser** [lammps/lammps#3313]
**Summary**

Running the in.swm4-ndp.nh example ends in a Fatal error in PMPI_Wait

**LAMMPS Version and Platform**

I found the issue using the 7 Jan 2022 version of LAMMPS. I updated to the 23 Jun 2022 version and got the same behavior. I compiled LAMMPS using cmake-3.22.2 with intel's 2020.4.304 compiler and intel's 2020.0.166 MPI. I used 8 processors on an intel ivybridge node that runs Red HAt Enterprise Linux-based Tri-Lab Operating System Stack.

**Expected Behavior**

I expected the simulation to run without trouble

**Actual Behavior**

Somewhere between steps 300 and 320 (the last thermo output was from step 300), the simulation crashed. The output file from the job said

Abort(269103118) on node 0 (rank 0 in comm 0): Fatal error in PMPI_Wait: Message truncated, error stack:
PMPI_Wait(208): MPI_Wait(request=0x7fffffffb9ac, status=0x1) failed
MPIR_Wait(108):
(unknown)(): Message truncated
Abort(805974030) on node 2 (rank 2 in comm 0): Fatal error in PMPI_Wait: Message truncated, error stack:
PMPI_Wait(208): MPI_Wait(request=0x7fffffffb99c, status=0x1) failed
MPIR_Wait(108):
(unknown)(): Message truncated
Abort(805974030) on node 4 (rank 4 in comm 0): Fatal error in PMPI_Wait: Message truncated, error stack:
PMPI_Wait(208): MPI_Wait(request=0x7fffffffb99c, status=0x1) failed
MPIR_Wait(108):
(unknown)(): Message truncated
Abort(805974030) on node 6 (rank 6 in comm 0): Fatal error in PMPI_Wait: Message truncated, error stack:
PMPI_Wait(208): MPI_Wait(request=0x7fffffffb99c, status=0x1) failed
MPIR_Wait(108):
(unknown)(): Message truncated

**Steps to Reproduce**

Try to run the provided examples/PACKAGES/drude/swm4-ndp/in.swm4-ndp.nh example using mpirun on 8 cores. I extended the run time to 1000000 steps, but that might have been unnecessary given it crashed between steps 300 and 320

**Further Information, Files, and Links**

**Comment by akohlmey** [comment]
@mdelyser thanks for your report. I can confirm the behavior although on my machine with my compiler and settings, it takes some more steps until this situation happens. It looks like there is something bad happening in the dynamics that doesn't always show and is perhaps now accelerated due to some changes to the core of LAMMPS. There is a significant number of "dangerous builds" in the reference logs already. That should not be and the neighbor list settings should be changed from the default. I suspect, some atoms go out of range and thus would be "lost" and then cannot be obtained for the communication required by fix rigid/small/npt.

A simple workaround for that is to use fix rigid/npt instead where this cannot happen.
But probably also the input should be updated with more conservative settings. E.g. the kspace setting of 1.0e-3 is far too aggressive for a variable cell calculation. something like 1.0-6 would be more appropriate. Also timestep could be chosen smaller and there could be an initial run with fix rigid/nvt/small instead of rigid/npt/small to avoid the large "pumping" of the box at the beginning due to the rather high potential energy and pressure of the initial geometry.

In conclusion, I suspect it is more the settings of the simulation leading to the communication issue than a bug in the code.  However, I am neither an expert on the DRUDE package nor on the communication patterns used by the rigid/.../small fixes. Thus I have assigned this to @sjplimp and @agiliopadua so they can provide more competent advice that I can.

**Comment by agiliopadua** [comment]
In the last weeks we found similar behaviour in systems mixing rigid/.../small with Drude dipoles. Instabilities in the drude integrators are common in systems with ions or highly polarizable atoms. Usually the symptoms are "Out of range atoms..." and we are used to handling those. But we hadn't seen MPI_Wait errors until now, in the new systems with rigid molecules. I'm not aware of recent changes in the Drude Nose-Hoover thermostats.

I'll update the input example and will try fix rigid/npt. But here in the group the same issue appeared in other systems.
